File: Data Structures & Algorithms/products-of-array-discluding-self/submission-3.py
class Solution:
    def productExceptSelf(self, nums: List[int]) -> List[int]:
        pre = [1] * len(nums)
        p = 1
        post = [1] * len(nums) 
        po = 1

        for i in range(len(nums)):
            p = pre[i] = p *nums[i]
        
        for i in range(len(nums)-1, -1, -1):
            po = post[i] = po * nums[i] 

        res = [1] * len(nums)
        for i in range(len(nums)):
            if i == 0:
                res[i] = post[i+1]
            elif i == len(nums) -1:
                res[i] = pre[i-1]
            else:
                res[i] = pre[i-1] * post[i+1]
        
        return res

File: Data Structures & Algorithms/two-integer-sum/submission-3.cpp
class Solution {
public:
    vector<int> twoSum(vector<int>& nums, int target) {
        int n = nums.size();
        vector<int> res;
        for(int i =0 ;i<n;i++){
            int second = target - nums[i];
            auto it = find(nums.begin(), nums.end(), second);
            if(it != nums.end() && it-nums.begin() != i){
                res.push_back(i);
                res.push_back(it-nums.begin());
                sort(res.begin(), res.end());
                return res;
            }
        }
        return res = {};
    }
};
